pkg/virtual/workspaces/registry/kubeconfig_rest_test.go

```diff
@@ -0,0 +1,887 @@
+			_, err := kubeconfigSubResourceStorage.Get(ctx, "foo", nil)
+			assert.EqualError(t, err, "workspaces/kubeconfig.tenancy.kcp.dev \"foo\" not found")
+			var statusError *kerrors.StatusError
+			require.ErrorAs(t, err, &statusError)
+			require.Len(t, statusError.Status().Details.Causes, 1)
+			assert.Equal(t, metav1.CauseTypeUnexpectedServerResponse, statusError.Status().Details.Causes[0].Type)
+			assert.Equal(t, "Key 'kubeconfig' not found in workspace shard Kubeconfig secret", statusError.Status().Details.Causes[0].Message)
+		},
+	}
+	applyTest(t, test)
+}
+
+func TestKubeconfigFailBecauseSecretNotFound(t *testing.T) {
+	user := &kuser.DefaultInfo{
+		Name:   "test-user",
+		UID:    "test-uid",
+		Groups: []string{"test-group"},
+	}
+	test := TestDescription{
+		TestData: TestData{
+			user:  user,
+			scope: organizationScope,
+			reviewerProvider: mockReviewerProvider{
+				"get":    mockReviewer{},
+				"delete": mockReviewer{},
+			},
+			workspaces: []tenancyv1alpha1.Workspace{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+					Status: tenancyv1alpha1.WorkspaceStatus{
+						BaseURL: "THE_RIGHT_SERVER_URL",
+						Location: tenancyv1alpha1.WorkspaceLocation{
+							Current: "theOneAndOnlyShard",
+						},
+						Conditions: conditionsv1alpha1.Conditions{
+							{
+								Type:   tenancyv1alpha1.WorkspaceURLValid,
+								Status: corev1.ConditionTrue,
+							},
+						},
+					},
+				},
+			},
+			workspaceShards: []tenancyv1alpha1.WorkspaceShard{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "theOneAndOnlyShard",
+					},
+					Spec: tenancyv1alpha1.WorkspaceShardSpec{
+						Credentials: corev1.SecretReference{
+							Name:      "kubeconfig",
+							Namespace: "kcp",
+						},
+					},
+				},
+			},
+			clusterRoleBindings: []rbacv1.ClusterRoleBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: getRoleBindingName(OwnerRoleType, "foo", user),
+						Labels: map[string]string{
+							PrettyNameLabel:   "foo",
+							InternalNameLabel: "foo",
+						},
+					},
+					Subjects: []rbacv1.Subject{
+						{
+							Kind: "User",
+							Name: user.Name,
+						},
+					},
+				},
+			},
+		},
+		apply: func(t *testing.T, storage *REST, kubeconfigSubResourceStorage *KubeconfigSubresourceREST, ctx context.Context, kubeClient *fake.Clientset, kcpClient *tenancyv1fake.Clientset, listerCheckedUsers func() []kuser.Info, testData TestData) {
+			_, err := kubeconfigSubResourceStorage.Get(ctx, "foo", nil)
+			assert.EqualError(t, err, "workspaces/kubeconfig.tenancy.kcp.dev \"foo\" not found")
+			var statusError *kerrors.StatusError
+			require.ErrorAs(t, err, &statusError)
+			require.Len(t, statusError.Status().Details.Causes, 1)
+			assert.Equal(t, metav1.CauseTypeUnexpectedServerResponse, statusError.Status().Details.Causes[0].Type)
+			assert.Equal(t, "secrets \"kubeconfig\" not found", statusError.Status().Details.Causes[0].Message)
```

Comment:
Should we really be writing end-to-end tests against status error messages? These are not part of our API.